python/ray/serve/_private/replica.py

```diff
@@ -303,16 +308,20 @@ async def is_allocated(self) -> str:
                 get_component_logger_file_path(),
             )
 
-        async def is_initialized(
+        async def initialized_and_get_metadata(
             self,
             deployment_config: DeploymentConfig = None,
             _after: Optional[Any] = None,
-        ):
+        ) -> Tuple[DeploymentConfig, DeploymentVersion]:
             # Unused `_after` argument is for scheduling: passing an ObjectRef
             # allows delaying reconfiguration until after this call has returned.
             try:
-                await self._initialize_replica()
-                metadata = await self.reconfigure(deployment_config)
+                async with self._replica_init_lock:
```

Comment:
Add a comment here for why we need the locking behavior (in what scenario this could be called twice).